service/worker/deployment/deployment_client.go

```diff
@@ -240,6 +250,47 @@ func (d *DeploymentClientImpl) DescribeDeployment(ctx context.Context, namespace
 	}, nil
 }
 
+func (d *DeploymentClientImpl) GetDeploymentReachability(
+	ctx context.Context,
+	namespaceEntry *namespace.Namespace,
+	seriesName string,
+	buildID string,
+) (*workflowservice.GetDeploymentReachabilityResponse, error) {
+	deployInfo, err := d.DescribeDeployment(ctx, namespaceEntry, seriesName, buildID)
+	if err != nil {
+		return nil, err
+	}
+	currentDeploymentInfo, err := d.GetCurrentDeployment(ctx, namespaceEntry, seriesName)
```

Comment:
I don't think you need to make both calls, GetCurrentDeployment has the info.

Reply:
deployInfo is the info of the deployment the user is asking about, which may or may not be the current deployment. I return `deployInfo` with `GetReachabilityResponse`, so that's why I get it. Honestly, I think it would maybe be better to remove `deployment info` from `GetReachabilityResponse`, and then I wouldn't have to do the `DescribeDeployment` call

Reply:
removed